pkg/operator/customroute/custom_route_controller.go

```diff
@@ -0,0 +1,283 @@
+			errors = append(errors, err)
+		}
+
+		if errors != nil {
+			return nil, "", errors
+		}
+
+		route.Spec.Host = hostname
+	}
+
+	return route, secretName, nil
+}
+
+func (c *customRouteController) validateCustomTLSSecret(secretName string) error {
+	if secretName != "" {
+		secret, err := c.secretLister.Secrets("openshift-config").Get(secretName)
+		if err != nil {
+			return err
+		}
+
+		var errors []error
+		privateKeyData, ok := secret.Data[corev1.TLSPrivateKeyKey]
+		if !ok {
+			errors = append(errors, fmt.Errorf("custom route secret must include key %s", corev1.TLSPrivateKeyKey))
+		} else {
+			errors = append(errors, certs.ValidatePrivateKey(privateKeyData)...)
+		}
+
+		certData, ok := secret.Data[corev1.TLSCertKey]
+		if !ok {
+			errors = append(errors, fmt.Errorf("custom route secret must include key %s", corev1.TLSCertKey))
+		} else {
+			errors = append(errors, certs.ValidateServerCert(certData)...)
+		}
+
+		if len(errors) != 0 {
+			return fmt.Errorf("error validating secret %s/%s: %v", "openshift-config", secretName, errors)
+		}
+	}
+	return nil
+}
+
+func (c *customRouteController) updateIngressConfigStatus(ctx context.Context, ingressConfig *configv1.Ingress, customRouteErrors []error) error {
+	// update ingressConfig status
+	// TODO: consider using c.targetRoute. Nevertheless the downside is that we might have a stale reference?
+	route, err := c.routeLister.Routes(c.targetRoute.ObjectMeta.Namespace).Get(c.targetRoute.ObjectMeta.Name)
+	if err != nil {
+		return err
+	}
+
+	componentRoute := &configv1.ComponentRouteStatus{
+		Namespace:        c.componentRoute.Namespace,
+		Name:             c.componentRoute.Name,
+		DefaultHostname:  configv1.Hostname(c.targetRoute.ObjectMeta.Name + "." + ingressConfig.Spec.Domain),
+		CurrentHostnames: []configv1.Hostname{configv1.Hostname(route.Spec.Host)},
+		ConsumingUsers:   c.consumingUsers,
+		RelatedObjects: []configv1.ObjectReference{
+			{
+				Group:     routev1.GroupName,
+				Resource:  "routes",
+				Name:      c.targetRoute.ObjectMeta.Name,
+				Namespace: c.targetRoute.ObjectMeta.Namespace,
+			},
+		},
+	}
+	conditions := checkErrorsConfiguringCustomRoute(customRouteErrors)
+	if conditions == nil {
+		conditions = checkIngressURI(ingressConfig, route)
+		if conditions == nil {
+			conditions = checkRouteAvailablity(c.secretLister, ingressConfig, route)
+		}
+	}
+	componentRoute.Conditions = ensureDefaultConditions(conditions)
+	_, err = c.updateComponentRouteStatus(ctx, componentRoute)
+	return err
+}
+
+func (c *customRouteController) syncSecret(secretName string) error {
+	source := resourcesynccontroller.ResourceLocation{}
+	if secretName != "" {
+		source = resourcesynccontroller.ResourceLocation{Namespace: "openshift-config", Name: secretName}
+	}
+
+	destination := resourcesynccontroller.ResourceLocation{Namespace: c.destSecret.Namespace, Name: c.destSecret.Name}
+
+	return c.resourceSyncer.SyncSecret(destination, source)
```

Comment:
this should use https://github.com/openshift/library-go/blob/14aa5b749d8ddbae509ea97c988228934dbbd3db/pkg/operator/resource/resourceapply/core.go#L387 instead

Reply:
Actually, it's pretty clever and I'd like to leave it :)

Doing it the way it is implemented we automatically handles deletes. If the `secretName` is empty, we'll create an empty `source` and pass it into the `resourceSyncer`. If the source is empty, the `resourceSyncer` is clever enough to delete the destination too. If I used the method you sent me, I'd need to duplicate this piece of code.

Reply:
fair enough :+1:

Reply:
Resolving.